File: src/main/java/com/weblearning/entity/Assignment.java
package com.weblearning.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "assignments")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Assignment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false, length = 100)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 125)
    private String attachmentUrl;

    @Column
    private LocalDateTime deadline;

    @Column(nullable = false)
    private Integer maxScore;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lesson_id", nullable = false)
    private Lesson lesson;

    @Column(nullable = false)
    private boolean deleted;

    @Column
    private LocalDateTime deletedAt;
}

File: frontend/src/types/assignment.ts
export interface Assignment {
  id: number
  lessonId: number
  title: string
  description: string | null
  attachmentUrl: string | null
  deadline: string | null
  maxScore: number
  createdAt: string
}

export interface CreateAssignmentPayload {
  title: string
  description?: string
  attachmentUrl?: string
  deadline?: string
  maxScore: number
}

export interface UpdateAssignmentPayload extends CreateAssignmentPayload {}
